x-pack/plugins/apm/server/routes/services/get_service_metadata_icons.ts

```diff
@@ -108,7 +108,12 @@ export async function getServiceMetadataIcons({
 
   let serverlessType: string | undefined;
   if (cloud?.provider === 'aws' && cloud?.service?.name === 'lambda') {
-    serverlessType = 'lambda';
+    serverlessType = 'aws.lambda';
```

Comment:
I think we could extract this to a function. I know we've done it in other places, but we could start by improving this new part 😅

```
function getServerlessType({cloudProvider, cloudServiceName}: {cloudProvider?: string, cloudServiceName?:string}){
if (cloud?.provider === 'aws' && cloud?.service?.name === 'lambda') {
    return ServerlessProvider.AWS...;
  } else if (
    cloud?.provider === 'azure' &&
    cloud?.service?.name === 'functions'
  ) {
    return ServerlessProvider.AZURE...;
  }
}
...
const serverlessType = getServerlessType(...)
  
```